Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm

```diff
@@ -324,7 +324,7 @@
     }
 }
 
-void RemoteLayerTreeDrawingAreaProxy::asyncSetLayerContents(WebCore::PlatformLayerIdentifier layerID, ImageBufferBackendHandle&& handle, const WebCore::RenderingResourceIdentifier& identifier)
+void RemoteLayerTreeDrawingAreaProxy::asyncSetLayerContents(WebCore::PlatformLayerIdentifier layerID, ImageBufferBackendHandle&& handle, const WebCore::ProcessQualified<WebCore::RenderingResourceIdentifier>& identifier)
```

Comment:
This one is a bit sad, admittedly, though hopefully can be fixed.

The UI process is keeping cached `CAIOSurface` objects for the three buffers, and needs to know which of these to keep/recycle/allocate each frame. It does so by tracking the `RenderingResourceIdentifier` associated with each, and the layer transaction provides the new three identifiers (plus one actual buffer handle) for it to match/update the cache.

Since the WebProcess no longer manages the three buffers, it doesn't have those identifiers to send. I made the GPUP send back the `BufferIdentifierSet` as part of the async reply just for this purpose. That set uses `ProcessQualified` (with the GPUP's process identity), since the identifiers were allocated in the GPUP, and we don't want them to collide with values allocated in the WebProcess (or in a different GPUP if the previous one crashes).

This code is using identifiers to resolve races between buffers delivered from `OffscreenCanvas` in a worker thread, and the same buffers being delivered through the main thread transaction, and doesn't really need the above. Unfortunately the main-thread path is using the same serialisation as the process qualified cache identifiers, so this changes to match.

It's a bit of sadness from RemoteLayerBackingStore representing quite different things (in-process 3 buffers, GPUP 3 buffers, or a delegated buffer) but all with a single serialization.

Maybe I'll just duplicate the identifier serialisation a bit rather than making this change.